src/librustc/ty/wf.rs

```diff
@@ -379,16 +379,21 @@ impl<'a, 'gcx, 'tcx> WfPredicates<'a, 'gcx, 'tcx> {
                     // obligations that don't refer to Self and
                     // checking those
 
-                    let cause = self.cause(traits::MiscObligation);
-                    let component_traits =
-                        data.auto_traits().chain(data.principal_def_id());
-                    self.out.extend(
-                        component_traits.map(|did| traits::Obligation::new(
-                            cause.clone(),
-                            param_env,
-                            ty::Predicate::ObjectSafe(did)
-                        ))
-                    );
+                    let defer_to_coercion =
```

Comment:
Note to self: think over the interaction of implied bounds here

Reply:
So I haven't really managed to think this over very much, though I did [ping @scalexm over on Zulip](https://rust-lang.zulipchat.com/#narrow/stream/144729-wg-traits/topic/object.20safe.20for.20dispatch.20.2357545) to start a conversation there (and I'd rather have that conversation there, or at least not on this PR). That said, I guess that we can land this PR, but we might have to roll it back.